source/change-streams/tests/unified/change-streams.yml

```
@@ -177,3 +177,163 @@ tests:
+            "a": 1
+          }
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult: {
+          "operationType": "addedInFutureMongoDBVersion",
+          "ns": {
+            "db": "database0",
+            "coll": "collection0"
+          }
+        }
+
+  - description: "Test newField added in response MUST NOT err"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: createChangeStream
+        object: *collection0
+        arguments:
+          # using $project to simulate future changes to ChangeStreamDocument structure
+          pipeline: [ { $project: { operationType: 1, ns: 1, newField: "newFieldValue" } } ]
+        saveResultAsEntity: &changeStream0 changeStream0
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: {
+            "_id": 1,
+            "a": 1
+          }
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult: {
+          "operationType": "insert",
+          "ns": {
+            "db": "database0",
+            "coll": "collection0"
+          },
+          "newField": "newFieldValue"
+        }
+
+  - description: "Test new structure in ns document MUST NOT err"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: createChangeStream
+        object: *collection0
+        arguments:
+          # using $project to simulate future changes to ChangeStreamDocument structure
+          pipeline: [ { $project: { operationType: "insert", "ns.viewOn": "db.coll" } } ]
+        saveResultAsEntity: &changeStream0 changeStream0
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: {
+            "_id": 1,
+            "a": 1
+          }
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult: {
+          "operationType": "insert",
+          "ns": {
+            "viewOn": "db.coll"
+          }
+        }
+
+  - description: "Test modified structure in ns document MUST NOT err"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: createChangeStream
+        object: *collection0
+        arguments:
+          # using $project to simulate future changes to ChangeStreamDocument structure
+          pipeline: [ { $project: { operationType: "insert", ns: { db: "$ns.db", coll: "$ns.coll", viewOn: "db.coll" } } } ]
+        saveResultAsEntity: &changeStream0 changeStream0
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: {
+            "_id": 1,
+            "a": 1
+          }
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult: {
+          "operationType": "insert",
+          "ns": {
+            "db": "database0",
+            "coll": "collection0",
+            "viewOn": "db.coll"
+          }
+        }
```

```suggestion
        expectResult:
          operationType: "insert",
          ns:
            db: *database0,
            coll: *collection0,
            viewOn: "db.coll"
```

The `ns` document was a little larger here, so I left it on multiple lines and just removed the redundant syntax.